torch/distributed/optim/named_optimizer.py

```diff
@@ -0,0 +1,258 @@
+import logging
+import warnings
+
+from copy import deepcopy
+from typing import Any, Collection, Dict, List, Mapping, Union
+
+import torch
+from torch import optim
+from torch.distributed._shard.sharded_tensor import ShardedTensor
+
+
+__all__ = ["_NamedOptimizer"]
```

Comment:
I am actually not sure if we should include `"_NamedOptimizer"` in `__all__` if we have it as private (with the preceding `_`).

Reply:
I believe it's the reason for test_correct_module_names failures.
Reverting this PR.

Reply:
Thanks... Sorry, I thought it was not caused by this pr... Should have read more carefully about the err msg..